addons/pos_sale/report/sale_report.py

```diff
@@ -53,14 +52,8 @@ def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
             partner.country_id AS country_id,
             partner.industry_id AS industry_id,
             partner.commercial_partner_id AS commercial_partner_id,
-            (select sum(t.weight*l.qty/u.factor) from pos_order_line l
-               join product_product p on (l.product_id=p.id)
-               left join product_template t on (p.product_tmpl_id=t.id)
-               left join uom_uom u on (u.id=t.uom_id)) AS weight,
-            (select sum(t.volume*l.qty/u.factor) from pos_order_line l
-               join product_product p on (l.product_id=p.id)
-               left join product_template t on (p.product_tmpl_id=t.id)
-               left join uom_uom u on (u.id=t.uom_id)) AS volume,
+            sum(p.weight * l.qty / u.factor) as weight,
+            sum(p.volume * l.qty / u.factor) as volume,
```

Comment:
Note: the resulting weight wasn't even valid because the template weight is 0 when it has multiple variants...